core/src/main/scala/com/raphtory/api/analysis/graphstate/GraphStateImplementation.scala

```diff
@@ -43,6 +43,60 @@ private object AccumulatorImplementation {
 
 import scala.math.Numeric.Implicits.infixNumericOps
 
+private class CounterImplementation[T] () extends Counter[T] {
+  var totalCount : Int = 0
+  var counts: mutable.Map[T, Int] = mutable.Map[T,Int]()
+
+  override def getCounts: mutable.Map[T, Int] = counts
+  override def largest : (T,Int) = counts.maxBy(_._2)
+  override def largest(k: Int): List[(T,Int)] = counts.toList.sortBy(_._2).takeRight(k.min(counts.size))
+
+  def +=(newValue: (T, Int)): Unit = this.synchronized{
+    if (counts.contains(newValue._1)) {
+      counts+=(newValue._1 -> (counts(newValue._1) + newValue._2))
+    } else {
+      counts+=(newValue._1 -> newValue._2)
+    }
+  }
+
+}
+
+private object CounterImplementation {
+  def apply[T]() = new CounterImplementation[T]()
+}
+
+private class CounterAccumulatorImplementation[T](retainState:Boolean) extends AccumulatorImplementation[(T, Int),Counter[T]] {
```

Comment:
Can you please add some unit tests to this class?
covering

- [ ] merge
- [ ] reset
- [ ] +=